### PR TITLE
feat: Add ability to read Agent configuration

### LINKED ISSUE #175: Add ability to inject custom configuration into default providers
Add ability to read configuration from yaml or json file.

example of aries configuration:

aries:
 agent:
   label: agent
   endpoint: https://example.com/endpoint

### PR DESCRIPTION
**Pull request by fqutishat**
- provide default config
- add ability to inject custom config

closes #175 

Signed-off-by: Firas Qutishat <firas.qutishat@securekey.com>

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+		if err != nil {
+			return nil, errors.Errorf("loading config file failed: %w", err)
+		}
+
+		return backend, nil
+	}
+}
+
+// FromRaw will initialize the configs from a byte array
+func FromRaw(configBytes []byte, configType string, opts ...Option) api.ConfigProvider {
```
[the remark this answers is not in the document]
Comment (fqutishat) [reply pkg/config/config.go]: ok

File: pkg/config/config.go [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+	err = backend.configViper.MergeConfig(in)
+	if err != nil {
+		return nil, err
+	}
+
+	return backend, nil
+}
+
+// WithEnvPrefix defines the prefix for environment variable overrides.
+// See viper SetEnvPrefix for more information.
```
Comment (llorllale) [inline pkg/config/config.go]: Suggest removing references to project dependencies in user-facing APIs. The godocs for user-facing APIs should explain themselves.

File: pkg/framework/aries/api/config.go [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+type Config interface {
+	AgentLabel() string
+	AgentServiceEndpoint() string
+}
+
+//ConfigProvider provides config backend for aries
+type ConfigProvider func() (ConfigBackend, error)
+
+//ConfigBackend backend for all config types in aries
+type ConfigBackend interface {
```
Comment (llorllale) [inline pkg/framework/aries/api/config.go]: I am confused by all this indirection around configuration. Why two interfaces? Why interfaces at all?

File: pkg/config/config.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (fqutishat) [reply pkg/config/config.go]: done

File: pkg/client/didexchange/client_test.go [hunk 4]
```diff
@@ -10,10 +10,12 @@ import (
 	"fmt"
 	"testing"
 
+	"github.com/hyperledger/aries-framework-go/pkg/framework/aries/api"
 	"github.com/stretchr/testify/require"
 
 	"github.com/hyperledger/aries-framework-go/pkg/didcomm/protocol/didexchange"
 	"github.com/hyperledger/aries-framework-go/pkg/didcomm/transport"
+
```
Comment (rolsonquadras) [inline pkg/client/didexchange/client_test.go]: extra line.

Comment (fqutishat) [reply pkg/client/didexchange/client_test.go]: done

File: pkg/config/config.go (hunk 2 above)
Comment (fqutishat) [reply pkg/config/config.go]: done

**Comment by fqutishat** [comment]
irrelevant now I will push new pr

**Pull request closed without merging**